src/prefect/server/database/migrations/env.py

```diff
@@ -126,7 +127,26 @@ def do_run_migrations(connection: AsyncEngine) -> None:
     )
 
     with context.begin_transaction():
-        context.run_migrations()
+        with disable_sqlite_foreign_keys(context):
+            context.run_migrations()
+
+
+@contextlib.contextmanager
+def disable_sqlite_foreign_keys(context):
+    """
+    Disable foreign key constraints on sqlite.
+    """
+    if dialect.name == "sqlite":
+        context.execute("COMMIT")
+        context.execute("PRAGMA foreign_keys=OFF")
+        context.execute("BEGIN")
```

Comment:
Foreign key behavior cannot be changed inside transactions so we cannot just disable them in migrations like we were before.

Reply:
Maybe this should be moved outside the `begin_transaction` block and we can skip the commit/begin 🤔